[alpha.webkit.UncountedCallArgsChecker] Check the safety of the object argument in a member function call.

rniwa opened the pull request:
This PR makes alpha.webkit.UncountedCallArgsChecker eplicitly check the safety of the object argument in a member function call. It also removes the exemption of local variables from this checker so that each local variable's safety is checked if it's used in a function call instead of relying on the local variable checker to find those since local variable checker currently has exemption for "for" and "if" statements.

haoNoQ commented on clang/lib/StaticAnalyzer/Checkers/WebKit/UncountedCallArgsChecker.cpp:
```diff
@@ -70,6 +70,15 @@ class UncountedCallArgsChecker
 
+      if (auto *MemberCallExpr = dyn_cast<CXXMemberCallExpr>(CE)) {
+        auto *E = MemberCallExpr->getImplicitObjectArgument();
+        auto *ArgType = MemberCallExpr->getObjectType().getTypePtrOrNull();
```
```suggestion
        QualType ArgType = MemberCallExpr->getObjectType();
```

`getTypePtrOrNull()` is typically redundant; `QualType` provides an overloaded `operator ->()` that allows you to call methods on the underlying unqualified `Type` directly.

rniwa replied:
Ah, okay. Will do that.

rniwa committed:
Address the review comment. Namely avoid calling getTypePtrOrNull and don't use raw_svector_ostream.

rniwa commented on clang/lib/StaticAnalyzer/Checkers/WebKit/UncountedCallArgsChecker.cpp:
```diff
@@ -183,6 +196,22 @@ class UncountedCallArgsChecker
+    SmallString<100> Buf;
+    llvm::raw_svector_ostream Os(Buf);
+
+    Os << "Call argument for 'this' parameter is uncounted and unsafe.";
```
Good point. Fixed.

haoNoQ reviewed (approved):
Great! Needs a rebase now though.

rniwa committed:
[alpha.webkit.UncountedCallArgsChecker] Check the safety of the object argument in a member function call.

This PR makes alpha.webkit.UncountedCallArgsChecker eplicitly check the safety of the object argument in
a member function call.

rniwa commented:
Rebased.

rniwa committed:
Fix the formatting.
